Utilities/Doxygen/DoxygenConfig.cmake

```
@@ -130,11 +130,11 @@ set(DOXYGEN_PREDEFINED
     "itkSimpleNewMacro(type)=             static Pointer New()\;"
     "itkCreateAnotherMacro(type)=         virtual::itk::LightObject::Pointer CreateAnother() const\;"
     "itkFactorylessNewMacro(x)=                   static Pointer New()\;                   virtual ::itk::LightObject::Pointer CreateAnother() const\;"
-    "itkTypeMacro(thisClass,superclass)=                   virtual const char *GetNameOfClass() const\;"
+    "itkOverrideGetNameOfClassMacro(thisClass)=                   virtual const char *GetNameOfClass() const\;"
     "itkVirtualGetNameOfClassMacro(thisClass)= virtual const char * GetNameOfClass() const\;"
     "itkOverrideGetNameOfClassMacro(thisClass)= const char * GetNameOfClass() const override\;"
     "itkEventMacro(thisClass,superclass)=                   class thisClass : public superclass {}\;"
-    "itkDeclareExceptionMacro(newexcp,parentexcp,whatmessage)=         namespace itk {         class newexcp : public parentexcp {        public:       newexcp(const char *file, unsigned int lineNumber) :        parentexcp(file, lineNumber)          { this->SetDescription(whatmessage)\; }          newexcp(const std::string & file, unsigned int lineNumber) :          parentexcp(file, lineNumber)          { this->SetDescription(whatmessage)\; }          itkTypeMacro(newexcp, , parentexcp)\;         }\;        }"
+    "itkDeclareExceptionMacro(newexcp,parentexcp,whatmessage)=         namespace itk {         class newexcp : public parentexcp {        public:       newexcp(const char *file, unsigned int lineNumber) :        parentexcp(file, lineNumber)          { this->SetDescription(whatmessage)\; }          newexcp(const std::string & file, unsigned int lineNumber) :          parentexcp(file, lineNumber)          { this->SetDescription(whatmessage)\; }          itkOverrideGetNameOfClassMacro(newexcp, , )\;         }\;        }"
```

```suggestion
    "itkDeclareExceptionMacro(newexcp,parentexcp,whatmessage)=         namespace itk {         class newexcp : public parentexcp {        public:       newexcp(const char *file, unsigned int lineNumber) :        parentexcp(file, lineNumber)          { this->SetDescription(whatmessage)\; }          newexcp(const std::string & file, unsigned int lineNumber) :          parentexcp(file, lineNumber)          { this->SetDescription(whatmessage)\; }          itkOverrideGetNameOfClassMacro(newexcp, )\;         }\;        }"
```
Should this semicolon be removed.

---

Which semicolon (`;`) do you mean?
Regarding the comma (`,`) in `itkOverrideGetNameOfClassMacro` also the second one in `itkOverrideGetNameOfClassMacro` should probably be removed .

---

I have no idea.  I will not be able to take this over the finish line.

---

Yes the second comma should also be removed as the macro in `Modules/Core/Common/include/itkMacro.h` doesn't have it either.

---

I agree with Albert. Removing those extra colons can be a second commit.